spec/services/book_serializer_spec.rb

```diff
@@ -2,50 +2,70 @@
 
 require 'rails_helper'
 
-RSpec.describe BookSerializer, type: :model do
+RSpec.describe BookSerializer, type: :service do
   let(:book1) do
-    create(:book, author_name: 'Michael Bond', title: 'Paddington Bear',
-                  description: 'The adventures of a small toy bear, Paddington after the train station.')
+    create(:book)
   end
   let(:book2) do
-    create(:book, author_name: 'A. A. Milne', title: 'Winnie the Pooh',
-                  description: 'A a good-natured, yellow-furred, honey-loving bear.', rating: 4, word_count: 600)
+    create(:book, author_name: 'A. A. Milne', title: 'Winnie the Pooh')
+  end
+
+  let(:expected_serialized_result) do
+    {
+      data: array_including(
+        a_hash_including(
+          type: 'books',
+          id: be_an_instance_of(String),
+          attributes: a_hash_including(
+            author_name: be_an_instance_of(String),
+            title: be_an_instance_of(String),
+            description: be_an_instance_of(String),
+            rating: be_an_instance_of(Integer).or(be_nil),
+            word_count: be_an_instance_of(Integer).or(be_nil)
+          )
+        )
+      )
+    }
+  end
+
+  let(:meta_result) do
+    {
+      meta:
+        a_hash_including(
+          status: be_an_instance_of(String),
+          message: be_an_instance_of(String)
+        )
+    }
   end
 
   context 'serialization of book' do
     let(:books) { [book1, book2] }
+    subject(:serialized_result) { BookSerializer.serialize(books) }
+
+    it 'returns serialized books with correct structure' do
+      expect(serialized_result).to include(expected_serialized_result)
+    end
 
-    it 'returns serialized books' do
-      serialized_result = BookSerializer.serialize(books)
-      expect(serialized_result).to eq(serialized_format)
+    it 'returns serialized books with correct meta message' do
+      expect(serialized_result).to include(meta_result)
     end
   end
-end
 
-def serialized_format
-  {
-    data: [book_data(book1, '1'), book_data(book2, '2')],
-    meta: {
-      status: 'SUCCESS',
-      message: 'Loaded books'
-    }
-  }
-end
+  context 'return of no books' do
+    let(:books) { [] }
+    subject(:serialized_result) { BookSerializer.serialize(books) }
 
-def book_data(book, id)
-  {
-    type: 'books',
-    id: id,
-    attributes: book_attributes(book)
-  }
-end
+    it 'returns nil when there are no books' do
+      expect(serialized_result).to be_nil
+    end
+  end
+
+  context 'when there is only 1 book' do
+    let(:books) { book1 }
+    subject(:serialized_result) { BookSerializer.serialize(books) }
 
-def book_attributes(book)
-  {
-    author_name: book.author_name,
-    title: book.title,
-    description: book.description,
-    rating: book.rating,
-    word_count: book.word_count
-  }
+    it 'returns 1 book' do
```

Comment:
This is not the expected result right?

This test is invalid it does not pass.

I understand you wanted to write a test for what the code should do.

But right now, I want you to write tests for what the code actually does. It raises a NoMethodError exception.

Can you write this test to assert what the behavior is please?

Then in the next MR we'll change this test to what it is here, alongside the change in the class to support the behavior we want.

Reply:
Could we go over this today? (Reminder comment)